main_test.go

```diff
@@ -461,7 +461,7 @@ func TestMatchFuzzyError(t *testing.T) {
 	t.Run("ambiguous migration: two different items have the same match", func(t *testing.T) {
 		create := set.NewStringSet(`abcde`, `abdecde`)
 		destroy := set.NewStringSet(`abdcde`, `hfjabd`)
-		wantError := "ambiguous migration: {abcde} -> {abdcde} or {abdecde} -> {abdcde}"
+		wantError := "ambiguous migration: map[abcde:abdcde abdecde:abdcde]"
```

Comment:
Let's take a step back: what it this PR fixing?

This PR is fixing a test flake, yes, but at the same time it is making the error message for the user less clear:

before this PR: 
`ambiguous migration: {abcde} -> {abdcde} or {abdecde} -> {abdcde}"`

with this PR:
`ambiguous migration: map[abcde:abdcde abdecde:abdcde]"`

Thus, we need another approach.

There are two possibilties:

1. we do the work in main.go to output an error message with the keys sorted AND still understandable by a human.
2. we don't touch main.go and instead we change the test code not to check for a string comparison, but instead to check for fragments, for example `strings.HasPrefix("ambiguous migration:")` and `strings.Contains("{abdcde}")` or similar.

Approach 1 would be the top, but for this particular case, approach 2 would be good enough, so I am fine with both.

Reply:
Done as suggested. I went with approach 2, because it seems to me that approach 1 could potentially bring more code and complicated logic for almost no benefit. The possibility of this `ambiguous migration` is incredibly low and the current prod/main code works well. As you wrote above the issue is because the test is flaky so I decided to address the issue there.